### DISCUSSION TITLE
how can i set an object to pinia state

### DISCUSSION THREAD

**Discussion by martin-matj** [vuejs/pinia#1393] [Help and Questions]
when i code like this. it do works. the data in pinia is updated

```
let thePerson = this.store.getPersonById(this.personId); //
thePerson.id_card = this.person.id_card;
thePerson.name = this.person.name;
thePerson.contact = this.person.contact;
thePerson.gender = this.person.gender;

```
but when i use any of these, it not work.
```
//thePerson = Object.assign({}, this.person);
//thePerson = JSON.parse(JSON.stringify(this.person));
//thePerson = { ...this.person }; not work
```
is there any short way to do this?

**Comment by posva** [comment]
if your store has a person property in the state, you can do `store.person = this.person`

**Reply by martin-matj** [reply]
it's a person list in my store. is there a similar method?

**Reply by posva** [reply]
you can use `store.persons.splice()` but you could also have an action that does that logic for you
